gspread/client.py

```diff
@@ -491,20 +491,26 @@ def insert_permission(
         """
 
         url = "{}/{}/permissions".format(DRIVE_FILES_API_V3_URL, file_id)
-
         payload = {
-            "emailAddress": value,
             "type": perm_type,
             "role": role,
             "withLink": with_link,
         }
-
         params = {
-            "sendNotificationEmail": notify,
-            "emailMessage": email_message,
             "supportsAllDrives": "true",
         }
 
+        if perm_type == "domain":
+            payload["domain"] = value
+        elif perm_type in {"user", "group"}:
+            payload["emailAddress"] = value
+            params["sendNotificationEmail"] = notify
+            params["emailMessage"] = email_message
+        elif perm_type == "anyone":
+            pass
+        else:
+            raise ValueError("Invalid permission type: {}".format(perm_type))
+
```

Comment:
IF this gets much more complicated then it should probably be refactored to somehow be delegated to a _get_user_group_permissions(), _get_domain_permissions(), _get_anyone_permissions() or similar

Reply:
I agree but so far from what I see in the drive API documentation that's all.